CitationStyleLanguagePlugin.inc.php

```diff
@@ -325,7 +339,49 @@ public function getArticleTemplateData($hookName, $args)
     }
 
     /**
-     * Get a specified citation for a given article
+     * Retrieve citation information for the preprint details template. This
+     * method is hooked in before a template displays.
+     *
+     * @see PreprintHandler::view()
+     *
+     * @param array $args
+     *
+     * @return false
+     */
+    public function getPreprintTemplateData($hookName, $args)
+    {
+        $request = $args[0];
+        $preprint = $args[1];
+        $publication = $args[2];
+        $context = $request->getContext();
+        $contextId = $context ? $context->getId() : 0;
+        $templateMgr = TemplateManager::getManager();
+
+        $citationArgs = [
+            'submissionId' => $preprint->getId(),
+            'publicationId' => $publication->getId(),
+        ];
+        $citationArgsJson = $citationArgs;
+        $citationArgsJson['return'] = 'json';
+
+        $templateMgr->assign([
+            'citation' => $this->getCitation($request, $submission, $this->getPrimaryStyleName($contextId), null, $publication),
```

Comment:
While testing I got a fatal error because `$submission` is undefined. I think you should change the `$preprint` variable name above to `$submission`.

Reply:
I changed it to `$preprint` since it was already declared and we use `$article` in the `getArticleTemplateData` method.